sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3102,7 +3103,12 @@ class Dataset[T] private[sql](
    * Collect all elements from a spark plan.
    */
   private def collectFromPlan(plan: SparkPlan): Array[T] = {
-    plan.executeCollect().map(boundEnc.fromRow)
+    val objProj = GenerateSafeProjection.generate(deserializer :: Nil)
```

Comment:
`fromRow` has caught `RuntimeException`. Shall we also catch it?

Reply:
it just rethrow the exception, not a big deal

Reply:
Ok. Looks good.